lib/msf/core/modules/metadata/search.rb

```diff
@@ -9,11 +9,61 @@ module Msf::Modules::Metadata::Search
       %w[aka author authors arch cve bid edb check date disclosure_date description fullname fullname mod_time
       name os platform path port rport rank ref ref_name reference references target targets text type]
 
+  #
+  # Module Type Shorthands
+  #
+  MODULE_TYPE_SHORTHANDS = {
+    "aux" => Msf::MODULE_AUX
+  }
+
+  #
+  # Parses command line search string into a hash. A param prefixed with '-' indicates "not", and will omit results
+  # matching that keyword. This hash can be used with the find command.
+  #
+  # Resulting Hash Example:
+  # {"platform"=>[["android"], []]} will match modules targeting the android platform
+  # {"platform"=>[[], ["android"]]} will exclude modules targeting the android platform
+  #
+  def self.parse_search_string(search_string)
+    search_string += ' '
```

Comment:
not sure why you're adding a space to the end of this only to split on spaces on the next line, it doesn't make a difference right?

Reply:
This was copied from the original implementation, I noticed this problem on the RPC service, with the input:

```
author:egypt arch:x64
```

This parsing method gives the following result:
```ruby
=> {"author"=>[["egypt arch:x64"], []]}
```

With the extra space appended, it gives the expected result of:

```ruby
=> {"author"=>[["egypt"], []], "arch"=>[["x64"], []]}
```

I'll add a unit test to make sure this scenario is caught correctly.

Reply:
Tests added. The tests were red without this line:

```
Failure/Error: it { expect(described_class.parse_search_string("author:egypt arch:x64")).to eq({"author"=>[["egypt"], []], "arch"=>[["x64"], []]}) }
     
       expected: {"arch"=>[["x64"], []], "author"=>[["egypt"], []]}
            got: {"author"=>[["egypt arch:x64"], []]}
     
       (compared using ==)
     
       Diff:
       @@ -1,3 +1,2 @@
       -"arch" => [["x64"], []],
       -"author" => [["egypt"], []],
       +"author" => [["egypt arch:x64"], []],
```

And now it's green with the line added back in :+1:

Reply:
https://github.com/rapid7/metasploit-framework/blob/9bd98f99424057aefa3ca61bfb25cc5b929b2cc8/lib/msf/ui/console/command_dispatcher/modules.rb#L496
isn't this where you got the code from? unless I'm missing something no space is being added to the search term

Reply:
The addition of the space is from the one that I'm unifying:

https://github.com/rapid7/metasploit-framework/blob/9bd98f99424057aefa3ca61bfb25cc5b929b2cc8/lib/msf/core/module/search.rb#L12-L16

Reply:
ah gotcha, that one has a guard clause, does that need added in here too? or are we stopping nil being passed in from elsewhere

Reply:
It's technically possible to provide `nil` via a message pack rpc call. I've updated this to handle nil, and added tests for an entirely empty string too. A nil search string will now result in empty search params, and thus return all modules.